File: DoAnChuyenNganh-main/DoAnCoSo/WebDuLichDaLat/Models/Region.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace WebDuLichDaLat.Models
{
    public class Region
    {
        [Column("RegionId")]
        public int Id { get; set; }

        [Column("RegionName")]
        [Required, StringLength(50)]
        public string Name { get; set; } = string.Empty;


        public ICollection<TouristPlace> TouristPlaces { get; set; } = new List<TouristPlace>();
    }

}

File: src/main/java/com/example/dalattravel/model/Region.java
package com.example.dalattravel.model;

import jakarta.persistence.*;
import lombok.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "regions")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Region {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "RegionId")
    private Integer id;

    @Column(name = "RegionName", nullable = false, length = 50)
    private String name;

    @OneToMany(mappedBy = "region", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    @Builder.Default
    private List<TouristPlace> touristPlaces = new ArrayList<>();
}
